packages/dev/core/src/Audio/sound.ts

```diff
@@ -201,6 +201,7 @@ export class Sound {
      */
     constructor(name: string, urlOrArrayBuffer: any, scene?: Nullable<Scene>, readyToPlayCallback: Nullable<() => void> = null, options?: ISoundOptions) {
         this.name = name;
+        this._urlOrArrayBuffer=urlOrArrayBuffer;

```

Comment:
```suggestion
        this._url = urlOrArrayBuffer;
```
... but move it down so it is right after `this._urlType = "String";` so it only gets set if the `urlOrArrayBuffer` argument is a string.